File: Data Structures & Algorithms/is-anagram/submission-8.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(t) != len(s):
            return False
        s_s = {}
        t_s = {}
        for ch in s:
            if ch not in s_s:
                s_s[ch] = 1
            s_s[ch] += 1
        for ch in t:
            if ch not in t_s:
                t_s[ch] = 1
            t_s[ch] += 1
        if t_s == s_s and len(t) == len(s) :
            return True
        return False

File: Data Structures & Algorithms/binary-tree-right-side-view/submission-8.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */

class Solution {
public:
    vector<int> rightSideView(TreeNode* root) {
        deque<TreeNode*> q;
        vector<int> result;
        int level_size;
        q.push_back(root);
        while(!q.empty()){
            level_size = q.size();
            for(int i=0;i<level_size;i++){
                TreeNode* node = q.front();
                q.pop_front();
                if(!node){
                    return result;
                }
                if(node->left){
                    q.push_back(node->left);
                }
                if(node->right){
                    q.push_back(node->right);
                }
                if(i == level_size - 1){
                    result.push_back(node->val);
                    
                }
            }
        }
        return result;
    }
};
